roles/openshift_hosted/tasks/registry.yml

```diff
@@ -10,7 +10,7 @@
 - name: set openshift_hosted facts
   set_fact:
     # This determines the gluster_ips to use for the registry by looping over the glusterfs_registry group
-    openshift_hosted_registry_storage_glusterfs_ips: "{%- set gluster_ips = [] %}{% if groups.glusterfs_registry is defined %}{% for node in groups.glusterfs_registry %}{%- set _ = gluster_ips.append(hostvars[node].glusterfs_ip | default(hostvars[node].openshift.common.ip)) %}{% endfor %}{{ gluster_ips }}{% elif groups.glusterfs is defined %}{% for node in groups.glusterfs %}{%- set _ = gluster_ips.append(hostvars[node].glusterfs_ip | default(hostvars[node].openshift.common.ip)) %}{% endfor %}{{ gluster_ips }}{% else %}{{ openshift_hosted_registry_storage_glusterfs_ips }}{% endif %}"
+    openshift_hosted_registry_storage_glusterfs_ips: "{%- set gluster_ips = [] %}{% if groups.glusterfs_registry is defined %}{% for node in groups.glusterfs_registry %}{%- set _ = gluster_ips.append(hostvars[node].glusterfs_ip | default(hostvars[node].openshift.common.ip if 'openshift' in hostvars[node])) %}{% endfor %}{{ gluster_ips }}{% elif groups.glusterfs is defined %}{% for node in groups.glusterfs %}{%- set _ = gluster_ips.append(hostvars[node].glusterfs_ip | default(hostvars[node].openshift.common.ip if 'openshift' in hostvars[node])) %}{% endfor %}{{ gluster_ips }}{% else %}{{ openshift_hosted_registry_storage_glusterfs_ips }}{% endif %}"
```

Comment:
This is a hideous unreadable block of jinja.  You really need to get in the habit of using vars: block and breaking down each little bit of logic into readable chunks, or write a filter in python or something.

Reply:
```yaml
 | default(hostvars[node].openshift.common.ip if 'openshift' in hostvars[node])
```
looks like it should have an else w/ some value.  Same with the other default value.

Reply:
When I initially wrote it it was nowhere near as bad, and no one told me about implementing through some other means on initial review. ;) I wouldn't mind making it easier to read, but that is a matter of aesthetics for a separate PR. Will update the defaults, however.